agent/grpc-external/services/resource/list_by_owner.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package resource
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/internal/storage"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+func (s *Server) ListByOwner(ctx context.Context, req *pbresource.ListByOwnerRequest) (*pbresource.ListByOwnerResponse, error) {
+	if err := validateListByOwnerRequest(req); err != nil {
+		return nil, err
+	}
+
+	_, err := s.resolveType(req.Owner.Type)
+	if err != nil {
+		return nil, err
+	}
+
+	childIds, err := s.Backend.OwnerReferences(ctx, req.Owner)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed getting owner references: %v", err)
+	}
+
+	authz, err := s.getAuthorizer(tokenFromContext(ctx))
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]*pbresource.Resource, 0)
+	for _, childId := range childIds {
+		reg, err := s.resolveType(childId.Type)
+		if err != nil {
+			return nil, err
+		}
+
+		// ACL filter
+		err = reg.ACLs.Read(authz, childId)
+		switch {
+		case acl.IsErrPermissionDenied(err):
+			continue
+		case err != nil:
+			return nil, status.Errorf(codes.Internal, "failed read acl: %v", err)
+		}
+
+		child, err := s.Backend.Read(ctx, readConsistencyFrom(ctx), childId)
```

Comment:
I think we should probably change the `Backend` to return full objects rather than IDs. I've taken a swing at it here https://github.com/hashicorp/consul/commit/ecd12b55e279f8263131beb7d4073e355faa9557 so you should be able to just cherry-pick it into your branch.

```shell
$ git fetch origin
$ git cherry-pick ecd12b55e279f8263131beb7d4073e355faa9557
```

Reply:
👍 Updated in https://github.com/hashicorp/consul/pull/17190/commits/cfff5a0f7ded9c0e60a921b2ce21596b72a298ca and https://github.com/hashicorp/consul/pull/17190/commits/c27a319b222d5a85bc05cf12e38bc42e59aea2f8